### PR TITLE
Visitor can read a current article

### PR DESCRIPTION
**Pull request by EevanR**
[PT](https://www.pivotaltracker.com/story/show/170676761)

- adds articles API controller with defined show
- adds article model with :title and :body
- adds rotues with format json
- rspec tests for json response and return article title
- article unit test

### CODE REVIEW

File: app/controllers/api/v1/articles_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Api::V1::ArticlesController < ApplicationController
+
+  def show
+    article = Article.all
```
Comment (oliverochman) [inline app/controllers/api/v1/articles_controller.rb]: this is a show action, not an index one

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
-end
+  namespace :api do
+    namespace :v1, defaults: { format: :json } do
+      resources :articles, only: [:show, :index]
```
Comment (oliverochman) [inline config/routes.rb]: no need for index here

File: spec/requests/api/v1/articles_spec.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  describe 'GET /api/v1/articles/:id' do
+    let!(:existing_entries) do
+      create(:article,
+            title: 'Breaking News',
+            body: 'Some breaking content')
+    end
+
+    before do
+      get '/api/v1/articles/:id', headers: headers
```
Comment (oliverochman) [inline spec/requests/api/v1/articles_spec.rb]: what id are you passing in here?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 2 above)
Comment (EevanR) [reply config/routes.rb]: done

File: spec/requests/api/v1/articles_spec.rb (hunk 3 above)
Comment (EevanR) [reply spec/requests/api/v1/articles_spec.rb]: changed

File: app/controllers/api/v1/articles_controller.rb (hunk 1 above)
Comment (EevanR) [reply app/controllers/api/v1/articles_controller.rb]: updated

File: spec/requests/api/v1/articles_spec.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+RSpec.describe Api::V1::ArticlesController, type: :request do
+  let(:headers) { { HTTP_ACCEPT: 'application/json' } }
+
+  describe 'GET /api/v1/articles/:id' do
+    let!(:existing_entries) do
+      create(:article,
+            id: 1,
```
Comment (oliverochman) [inline spec/requests/api/v1/articles_spec.rb]: remove id from here

File: spec/requests/api/v1/articles_spec.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+            title: 'Breaking News',
+            body: 'Some breaking content')
+    end
+
+    before do
+      get '/api/v1/articles/1', headers: headers
```
Comment (oliverochman) [inline spec/requests/api/v1/articles_spec.rb]: and pass in `article.id` here

File: spec/requests/api/v1/articles_spec.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+RSpec.describe Api::V1::ArticlesController, type: :request do
+  let(:headers) { { HTTP_ACCEPT: 'application/json' } }
+
+  describe 'GET /api/v1/articles/:id' do
+    let!(:existing_entries) do
```
Comment (oliverochman) [inline spec/requests/api/v1/articles_spec.rb]: I would change this to `article`

File: app/controllers/api/v1/articles_controller.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+class Api::V1::ArticlesController < ApplicationController
+
+  def show
+    article = Article.find(params[:id])
+    render json: { entries: article }
```
Comment (oliverochman) [inline app/controllers/api/v1/articles_controller.rb]: I do not think it makes sense to have the article inside of a key called `entries`. 
Can you have `render json: article`

File: spec/requests/api/v1/articles_spec.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+    it 'returns an article' do
+      expect(response_json.count).to eq 1
+    end
+
+    it 'returns article title' do
+      expect(response_json["entries"]["title"]).to eq "Breaking News"
+    end
+
+  end
```
Comment (oliverochman) [inline spec/requests/api/v1/articles_spec.rb:32]: what happens with a request when there is no article with the `id` being sent?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: spec/requests/api/v1/articles_spec.rb (hunk 4 above)
Comment (EevanR) [reply spec/requests/api/v1/articles_spec.rb]: ok

File: spec/requests/api/v1/articles_spec.rb (hunk 5 above)
Comment (EevanR) [reply spec/requests/api/v1/articles_spec.rb]: ok

File: spec/requests/api/v1/articles_spec.rb (hunk 6 above)
Comment (EevanR) [reply spec/requests/api/v1/articles_spec.rb]: ok

File: app/controllers/api/v1/articles_controller.rb (hunk 7 above)
Comment (EevanR) [reply app/controllers/api/v1/articles_controller.rb]: yes

File: spec/requests/api/v1/articles_spec.rb (hunk 8 above)
Comment (EevanR) [reply spec/requests/api/v1/articles_spec.rb:32]: sad path has been put in place

File: app/controllers/api/v1/articles_controller.rb [hunk 9]
```diff
@@ -0,0 +1,13 @@
+class Api::V1::ArticlesController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
+
+  def show
+    article = Article.find(params[:id])
+    render json: article 
+  end
+
+  def not_found
```
Comment (oliverochman) [inline app/controllers/api/v1/articles_controller.rb:11]: move this to the application controller and make sure it is a private method

File: spec/requests/api/v1/articles_spec.rb [hunk 10]
```diff
@@ -0,0 +1,33 @@
+RSpec.describe Api::V1::ArticlesController, type: :request do
```
Comment (oliverochman) [inline spec/requests/api/v1/articles_spec.rb]: this should be described as `"GET /api/v1/articles/:id"`

File: spec/requests/api/v1/articles_spec.rb [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+  let(:headers) { { HTTP_ACCEPT: 'application/json' } }
+  let!(:article) do
+    create(:article,
+          title: 'Breaking News',
+          body: 'Some breaking content')
+  end
+
+  describe 'GET /api/v1/articles/:id' do
```
Comment (oliverochman) [inline spec/requests/api/v1/articles_spec.rb]: change to "Successfully"

File: spec/requests/api/v1/articles_spec.rb [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+    it 'returns a 200 response status' do
+      expect(response).to have_http_status 200
+    end
+
+    it 'returns article title' do
+      expect(response_json["title"]).to eq "Breaking News"
+    end
+  end
+
+  describe 'GET /api/v1/articles/:id' do
```
Comment (oliverochman) [inline spec/requests/api/v1/articles_spec.rb]: change to "With invalid :id"

File: app/controllers/api/v1/articles_controller.rb [hunk 13]
```diff
@@ -0,0 +1,13 @@
+class Api::V1::ArticlesController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
+
+  def show
+    article = Article.find(params[:id])
+    render json: article
```
Comment (oliverochman) [inline app/controllers/api/v1/articles_controller.rb]: please serialize response

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/articles_controller.rb (hunk 9 above)
Comment (EevanR) [reply app/controllers/api/v1/articles_controller.rb:11]: ok

File: spec/requests/api/v1/articles_spec.rb (hunk 11 above)
Comment (EevanR) [reply spec/requests/api/v1/articles_spec.rb]: ok

File: spec/requests/api/v1/articles_spec.rb (hunk 12 above)
Comment (EevanR) [reply spec/requests/api/v1/articles_spec.rb]: ok

File: spec/requests/api/v1/articles_spec.rb (hunk 10 above)
Comment (EevanR) [reply spec/requests/api/v1/articles_spec.rb]: ok

File: app/controllers/api/v1/articles_controller.rb (hunk 13 above)
Comment (EevanR) [reply app/controllers/api/v1/articles_controller.rb]: ok

**Review by oliverochman — APPROVED** [review]

**Pull request merged**